py/picca/utils.py

```diff
@@ -128,47 +128,67 @@ def desi_from_truth_to_drq(truth,targets,drq,spectype="QSO"):
 
     return
 
-def desi_from_ztarget_to_drq(ztarget,drq,spectype="QSO"):
-    '''
-    Transform a desi truth.fits file and a
-    desi targets.fits into a drq like file
+def desi_from_ztarget_to_drq(zcat,drq,spectype='QSO',downsampling_z_cut=None, downsampling_nb=None):
+    """Transforms a catalog of object in desi format to a catalog in DRQ format
 
-    '''
+    Args:
+        zcat (str): path to the catalog of object
+        drq (str): path to write the DRQ catalog
+        spectype (str): Spectype of the object, can be any spectype
+            in desi catalog. Ex: 'STAR', 'GALAXY', 'QSO'
+        downsampling_z_cut (float) : Minimum redshift to downsample
+            the data, if 'None' no downsampling
+        downsampling_nb (int) : Target number of object above redshift
+            downsampling-z-cut, if 'None' no downsampling
 
-    vac = fitsio.FITS(ztarget)
+    Returns:
+        None
+
+    """
 
     ## Info of the primary observation
-    thid  = vac[1]["TARGETID"][:]
-    ra    = vac[1]["RA"][:]
-    dec   = vac[1]["DEC"][:]
-    zqso  = vac[1]["Z"][:]
-    plate = 1+sp.arange(thid.size)
-    mjd   = 1+sp.arange(thid.size)
-    fid   = 1+sp.arange(thid.size)
-    sptype = sp.char.strip(vac[1]["SPECTYPE"][:].astype(str))
+    vac = fitsio.FITS(ztarget)
+    sptype = sp.char.strip(vac[1]['SPECTYPE'][:].astype(str))
 
     ## Sanity
-    print(" start               : nb object in cat = {}".format(ra.size) )
-    w = (vac[1]["ZWARN"][:]==0.)
-    print(" and zwarn==0        : nb object in cat = {}".format(ra[w].size) )
-    w = w & (sptype==spectype)
-    print(" and spectype=={}    : nb object in cat = {}".format(spectype,ra[w].size) )
-
-    ra    = ra[w]
-    dec   = dec[w]
-    zqso  = zqso[w]
-    thid  = thid[w]
-    plate = plate[w]
-    mjd   = mjd[w]
-    fid   = fid[w]
+    w = sp.ones(ra.size).astype(bool)
```

Comment:
it's more efficient to instantiate a boolean dorectly rather than casting from float:

`w = sp.ones(ra.size, dtype=bool)`